### ISSUE TITLE
Crash after splash [2017.1, GVR 1.7, iOS XCode 8.3.3]

### ISSUE THREAD

**Issue by FuzzyWuzzie** [googlevr/gvr-unity-sdk#665]
Can't get anything to run past the splash screen; it just crashes immediately here:

```
extern "C"  void GvrAudio_Initialize_m2441973946 (RuntimeObject * __this /* static, unused */, int32_t ___quality0, int32_t ___sampleRate1, int32_t ___numChannels2, int32_t ___framesPerBuffer3, const RuntimeMethod* method)
{
	typedef void (DEFAULT_CALL *PInvokeFunc) (int32_t, int32_t, int32_t, int32_t);

	// Native function invocation
	reinterpret_cast<PInvokeFunc>(Initialize)(___quality0, ___sampleRate1, ___numChannels2, ___framesPerBuffer3); // crashes here with: Thread 1: EXC_BAD_ACCESS (code=1, address=0x9000000000000000)

}
```

```
Thread 1Queue : com.apple.main-thread (serial)
#0	0x000000018468d034 in __bzero ()
#1	0x0000000100f85a30 in vraudio::Resampler::Resampler() ()
#2	0x0000000100f81b6c in vraudio::GraphManager::GraphManager(vraudio::GraphManager::RenderingMode, vraudio::SystemSettings const&) ()
#3	0x0000000100f7de68 in vraudio::VrAudioApiImpl::VrAudioApiImpl(vraudio::VrAudioApi::RenderingMode, unsigned long, unsigned long, int) ()
#4	0x000000010059defc in vraudio::VrAudioApi::Create(vraudio::VrAudioApi::RenderingMode, unsigned long, unsigned long, int) ()
#5	0x0000000100f7a16c in Initialize ()
#6	0x0000000100073bc8 in GvrAudio_Initialize_m2441973946 [inlined] at /Users/kenton/Projects/Build/Classes/Native/Bulk_Assembly-CSharp_0.cpp:29157
#7	0x0000000100073bc0 in ::GvrAudio_Initialize_m2739768702(RuntimeObject *, GvrAudioListener_t1521766837 *, int32_t, const RuntimeMethod *) at /Users/kenton/Projects/Build/Classes/Native/Bulk_Assembly-CSharp_0.cpp:28256
#8	0x0000000100444cb4 in RuntimeInvoker_Void_t1841601450(void (*)(), MethodInfo const*, void*, void**) at /Users/kenton/Projects/Build/Classes/Native/Il2CppInvokerTable.cpp:19617
#9	0x0000000100f70d3c in il2cpp::vm::Runtime::Invoke(MethodInfo const*, void*, void**, Il2CppException**) at /Users/builduser/buildslave/unity/build/External/il2cpp/il2cpp/libil2cpp/vm/Runtime.cpp:464
#10	0x0000000100a110c8 in ::scripting_method_invoke() at /Users/builduser/buildslave/unity/build/Runtime/ScriptingBackend/Il2Cpp/ScriptingApi_Il2Cpp.cpp:218
#11	0x0000000100a0df68 in ::Invoke() at /Users/builduser/buildslave/unity/build/Runtime/Scripting/ScriptingInvocation.cpp:303
#12	0x0000000100a0dea0 in Invoke [inlined] at /Users/builduser/buildslave/unity/build/Runtime/Scripting/ScriptingInvocation.h:72
#13	0x0000000100a0de94 in ::InvokeChecked() at /Users/builduser/buildslave/unity/build/Runtime/Scripting/ScriptingInvocation.cpp:128
#14	0x00000001009b1734 in ::CallMethodInactive() at /Users/builduser/buildslave/unity/build/Runtime/Mono/MonoBehaviour.cpp:330
#15	0x00000001009b3748 in ::CallAwake() at /Users/builduser/buildslave/unity/build/Runtime/Mono/MonoBehaviour.cpp:1301
#16	0x00000001009b3ab0 in ::AddToManager() at /Users/builduser/buildslave/unity/build/Runtime/Mono/MonoBehaviour.cpp:1410
#17	0x00000001009b36a8 in ::AwakeFromLoad() at /Users/builduser/buildslave/unity/build/Runtime/Mono/MonoBehaviour.cpp:1282
#18	0x0000000100a12450 in ::InvokePersistentManagerAwake() at /Users/builduser/buildslave/unity/build/Runtime/Serialize/AwakeFromLoadQueue.cpp:391
#19	0x0000000100a12354 in ::PersistentManagerAwakeFromLoad() at /Users/builduser/buildslave/unity/build/Runtime/Serialize/AwakeFromLoadQueue.cpp:277
#20	0x0000000100a122b0 in ::PersistentManagerAwakeFromLoad() at /Users/builduser/buildslave/unity/build/Runtime/Serialize/AwakeFromLoadQueue.cpp:150
#21	0x0000000100a01484 in ::CompleteAwakeSequence() at /Users/builduser/buildslave/unity/build/Runtime/PreloadManager/LoadSceneOperation.cpp:563
#22	0x0000000100a014d0 in ::CompletePreloadManagerLoadScene() at /Users/builduser/buildslave/unity/build/Runtime/PreloadManager/LoadSceneOperation.cpp:576
#23	0x0000000100a010c8 in ::PlayerLoadSceneFromThread() at /Users/builduser/buildslave/unity/build/Runtime/PreloadManager/LoadSceneOperation.cpp:486
#24	0x0000000100a011a8 in ::CompleteLoadFirstScene() at /Users/builduser/buildslave/unity/build/Runtime/PreloadManager/LoadSceneOperation.cpp:517
#25	0x0000000100a00e00 in ::IntegrateMainThread() at /Users/builduser/buildslave/unity/build/Runtime/PreloadManager/LoadSceneOperation.cpp:281
#26	0x0000000100a01c70 in ::UpdatePreloadingSingleStep() at /Users/builduser/buildslave/unity/build/Runtime/PreloadManager/PreloadManager.cpp:475
#27	0x0000000100a02300 in ::UpdatePreloading() at /Users/builduser/buildslave/unity/build/Runtime/PreloadManager/PreloadManager.cpp:613
#28	0x0000000100997228 in ::PlayerLoop() at /Users/builduser/buildslave/unity/build/Runtime/Misc/PlayerLoop.cpp:103
#29	0x0000000100b53110 in ::UnityPlayerLoopImpl() at /Users/builduser/buildslave/unity/build/PlatformDependent/iPhonePlayer/LibEntryPoint.mm:221
#30	0x0000000100051fc0 in ::UnityRepaint() at /Users/kenton/Projects/Build/Classes/UnityAppController+Rendering.mm:268
#31	0x0000000100051eac in ::-[UnityAppController(Rendering) repaintDisplayLink]() at /Users/kenton/Projects/Build/Classes/UnityAppController+Rendering.mm:77
#32	0x00000001030684fc in __clang_call_terminate ()
#33	0x00000001887e45dc in CA::Display::DisplayLinkItem::dispatch(unsigned long long) ()
#34	0x00000001887e448c in CA::Display::DisplayLink::dispatch_items(unsigned long long, unsigned long long, unsigned long long) ()
#35	0x000000018583fb9c in IODispatchCalloutFromCFMessage ()
#36	0x0000000185569960 in __CFMachPortPerform ()
#37	0x0000000185581ae4 in __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ ()
#38	0x0000000185581284 in __CFRunLoopDoSource1 ()
#39	0x000000018557ed98 in __CFRunLoopRun ()
#40	0x00000001854aeda4 in CFRunLoopRunSpecific ()
#41	0x0000000186f18074 in GSEventRunModal ()
#42	0x000000018b769058 in UIApplicationMain ()
#43	0x000000010004bafc in main at /Users/kenton/Projects/Build/Classes/main.mm:35
#44	0x00000001844bd59c in start ()
```

**Comment by FuzzyWuzzie** [comment]
I was able to "fix" this by deleting all Google VR audio stuff (scripts, monobehaviours, and libraries) from the project since I don't need spatial audio anyway.

**Comment by VasilOrlinov** [comment]
I have similar issues with 2017.2, GVR 1.7. When i try to use GVR spatial audio my Unity project crashes every time.

**Comment by CenekStrichel** [comment]
Same trouble. It looks that Google VR is absolutely unusable on iOS in current version and with current Unity.

**Comment by GoGoBlitz** [comment]
Any news? Also have crash immediately after game starts with 2017 unity + gvr 1.7

**Comment by diegodias** [comment]
Same here. Cant even get the demos to work.

**Comment by bunnyhero** [comment]
@FuzzyWuzzie thanks for the workaround!

fwiw, this is what i deleted from my project to get it to run without crashing (unsure if i needed to delete all of these or not, but):

- everything in __GoogleVR/Editor/Audio__
- everything in __GoogleVR/Plugins/iOS__
- __GoogleVR/Resources/GvrAudioMixer.mixer__
- everything in __GoogleVR/Scripts/Audio__

**Comment by rusmaxham** [comment]
A GVR Audio engineer is looking into this now.

**Comment by miraclestar** [comment]
Same error

**Comment by tribuanimal** [comment]
Thanks a lot for the solutions. Worked perfectly.

**Comment by soundreactor** [comment]
same here

**Comment by GoGoBlitz** [comment]
Well, I recommend to all downgrade to unity 5.5.3 + gvr 1.4 it is only combo which works perfectly without bugs and with good fps.

**Comment by zhuyifan2013** [comment]
same here!! This issue is very serious !

**Comment by biezhihua** [comment]
same here!! This issue is very serious !

**Comment by anokta** [comment]
Hi,

We have tracked down the issue and will provide a fix from in the next SDK update. In the meantime, [here is a workaround patch to resolve the issue](https://gist.github.com/anokta/87e0b440138c921664806ba6e9bd3cfd). To apply the patch on Mac:

1. Open your Unity project, and build a new iOS app as usual - which should generate the corresponding Xcode project.
2. Close the Xcode project (if open already), and make sure you backup its contents, as the scripts will directly modify the library & project files in the *Pods* folder.
3. Download and copy the attached script into the root folder of the generated Xcode project, i.e., the folder that contains the *.xcodeproj* file and the *Pods* folder (will be referred as $XCODE_PROJ_FOLDER below).
4. Open Terminal, then run the following:
```bash
cd $XCODE_PROJ_FOLDER
chmod +x iosunityfix.sh
./iosunityfix.sh
```
5. Once finished, open the Xcode project to build and run the iOS app. The new app should run as expected without any crashes now.

Please note that running this patch is required **once** everytime you generate a *new* Xcode project through Unity.

**Comment by rusmaxham** [comment]
This should be fixed with the most recent versions of Unity 5.6 and 2017.1+ (5.6.4f1 and 2017.1.2f1).

**Comment by rusmaxham** [comment]
Alternatively, following the [Pod update to the latest GVR SDK for iOS](https://github.com/googlevr/gvr-unity-sdk#pod-update-to-the-latest-gvr-sdk-for-ios) instructions to update to a newer lib GVR will also fix this issue if you absolutely cannot upgrade your version of Unity.
